pkg/cloud/gcpprovider.go

```diff
@@ -795,12 +804,28 @@ func (gcp *GCP) parsePage(r io.Reader, inputKeys map[string]Key, pvKeys map[stri
 					if gpuType != "" {
 						lastRateIndex := len(product.PricingInfo[0].PricingExpression.TieredRates) - 1
 						var nanos float64
+						var unitsBaseCurrency int
 						if lastRateIndex > -1 && len(product.PricingInfo) > 0 {
 							nanos = product.PricingInfo[0].PricingExpression.TieredRates[lastRateIndex].UnitPrice.Nanos
+							unitsBaseCurrency, err = strconv.Atoi(product.PricingInfo[0].PricingExpression.TieredRates[lastRateIndex].UnitPrice.Units)
+							if err != nil {
+								return nil, "", err
+							}
 						} else {
 							continue
 						}
-						hourlyPrice := nanos * math.Pow10(-9)
+
+						// as per https://cloud.google.com/billing/v1/how-tos/catalog-api
+						// the hourly price is the whole currency price + the fractional currency price
+						hourlyPrice := (nanos * math.Pow10(-9)) + float64(unitsBaseCurrency)
+
+						// GPUs with an hourly price of 0 are reserved versions of GPUs
+						// (E.g., SKU "2013-37B4-22EA")
+						// and are excluded from cost computations
+						if hourlyPrice < 0.001 {
```

Comment:
Is there any special significance to `0.001`?

Reply:
I am using this like an epsilon value to prevent some dust at the end of the float making it seem like this is an actual value. But the fact that this causes more questions makes it probably not the most maintainable choice. I have not _observed_ this behavior so this might be over designed. 

Going to do a straight equality with 0

Reply:
I see the merit of comparing against an epsilon value, especially with the earlier float math. I was mainly asking if 0.001 had significance in particular; if you'd like to keep the epsilon go for it! All I'd ask is a note explaining exactly what you just explained to me 😁